blob/blob.go

```diff
@@ -0,0 +1,39 @@
+package blob
+
+import (
+	"context"
+	"errors"
+	"io"
+
+	"github.com/ipfs/go-cid"
+)
+
+var (
+	ErrBlobTooLarge = errors.New("blob size exceeds the maximum allowed")
+	ErrBlobNotFound = errors.New("no blob is found with given ID")
+)
+
+type (
+	ID         cid.Cid // TODO: Discuss if everyone is on board with using CIDs as blob ID.
```

Comment:
the place this will break down is on partial / really large uploads - we don't know the cid until we get all the data / chunk it.
so how do we make a reference for something in order to stream it in

Reply:
The maximum supported upload size is currently 32GiB. How I imagine this would work is that data is stored locally (or offloaded to a remote staging area like S3 or R2), then the deal making starts. For the deal making we need the CID of the blob anyway I _think?_. And the ID is not returned until POST /v0/blob finishes. 

Can you help me understand in what specific scenario we might need the reference before POST /v0/blob ends?

Reply:
we aren't supporting larger uploads yet., but we may need to in the future.
is there a reasonable way to future proof ourselves to support that?

Reply:
(should we just allocate uuids as references, which we'll always be able to do)

Reply:
Maybe. 

From client's perspective IDs remain opaque string. The fact that they are cids in this PR should remain an internal implementation detail. 

I'll change this to UUIDs so that nothing could go wrong in the meantime. Thanks Will